src/GitServer/pack-objects/GitObjectManager.ts

```diff
@@ -0,0 +1,81 @@
+import pako from 'pako'
+import path from 'path'
+import fs from 'fs'
+
+import GitObject from './GitObject'
+import { GitPackIndex } from './GitPackIndex'
+import { EncryptedFS } from 'encryptedfs'
+
+const PackfileCache = new Map()
+
+class GitObjectManager {
```

Comment:
With this amount of complexity, I'm hoping there should be tests for all of this.

Reply:
I will be reducing complexity to cater to our specific needs as well as including some tests as a part of this PR

Reply:
I've been able to reduce complexity a little within the imported code but there is still a lot there just due to the fact that we are handling the git backend on the lowest level (directly on the git protocol level with network packets). It would be great if there was a library that could handle all this but to the best of my knowledge, a git backend in pure JavaScript has never been done in production level code.

Reply:
I am adding several tests on the macro level (vaults pulling from peer vaults) and I think this might be enought for now just have inital confidence that it will work the way we intended but more comprehensive testing is definitely needed for the git backend code (`packObjects`, `GitSideBand` and `uploadPack`). I plan on diving deep into all of that in future so we can really make that code our own and not just a copy from isomorphic-git.